presto-main/src/main/java/com/facebook/presto/type/DecimalCasts.java

```diff
@@ -495,8 +495,13 @@ public static Slice doubleToLongDecimal(double value, long precision, long scale
 
     private static Slice internalDoubleToLongDecimal(double value, long precision, long scale)
     {
+        if (Double.isInfinite(value) || Double.isNaN(value)) {
```

Comment:
cmt msg
> Make double/real->decimal cast compatible with cast to varchar ...

`s/discourage/&d`
`s/as s/as the s`
`s/change/&,`
`s/naturally prints out/is printed`
`s/out/as`
`s/when/and &`